tests/atoms/gaia_data_layer.js

```diff
@@ -66,16 +67,17 @@ var GaiaDataLayer = {
 
   enableBluetooth: function() {
     var bluetooth = window.navigator.mozBluetooth;
-    if (!bluetooth.enabled) {
+    if (bluetooth.defaultAdapter.state != 'enabled') {
```

Comment:
So I'm guessing, when bluetooth is being enabled, it goes through the transition of disabled -> enabling -> enabled.  and enabled-> disabling -> disabled the other way.  So if it is in middle of enabling, (but not enabled yet)  we should still wait until it is enabled, then try to disable it in case of line 48.  So if we only check for != 'enabled' state, then it will ignore the disable request of the user.  

I'm wondering whether that actually makes sense or not.  should we ignore the input given during the transition state?  I'm leaning towards no, but you guys might already have reached conclusion since this comment was made 12 hours ago.